harness/determined/searcher/search_method.py

```diff
@@ -182,18 +206,42 @@ def on_trial_exited_early(
         """
         pass
 
-    def save_checkpoint(self, event_id: int) -> None:
+    def save(self, path: pathlib.Path, *, experiment_id: int, event_id: int) -> None:
         """
-        This is optionally implemented to save a checkpoint indexed by event id.
+        This is optionally overridden to save a checkpoint indexed by event id.
         It will be called by the ``SearchRunner`` after receiving operations
         from the ``SearchMethod``
         """
+        searcher_state_file = path.joinpath(STATE_FILE)
+        self.save_method_state(path, event_id)
+        d = self.searcher_state.to_dict()
+        d["experimentId"] = experiment_id
+        with searcher_state_file.open("w") as f:
+            json.dump(d, f)
+
+    def save_method_state(self, path: pathlib.Path, event_id: int) -> None:
```

Comment:
thanks for making this api so much better!  much cleaner to handle shared searcher state completely and have the user just focus on the method state.  also user does not need to worry about event_id anymore as well.